hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/response/volume/OMVolumeSetOwnerResponse.java

```diff
@@ -55,11 +56,27 @@ public OMVolumeSetOwnerResponse(@Nonnull OMResponse omResponse,
 
   /**
    * For when the request is not successful or it is a replay transaction.
```

Comment:
should we remove this comment as checkStatusNotOK is now conditioned with success=true? Also, do we redirect existing callers for success=false case appropriately?

Reply:
I think the reply response is taken care by createReplayOMResponse now. So we can safely clean up the comments above.

Reply:
I believe the comment is still accurate. There are four places that calls this constructor at the moment:

- INVALID_REQUEST
```java
    // In production this will never happen, this request will be called only
    // when we have ownerName in setVolumePropertyRequest.
    if (!setVolumePropertyRequest.hasOwnerName()) {
      omResponse.setStatus(OzoneManagerProtocolProtos.Status.INVALID_REQUEST)
          .setSuccess(false);
      return new OMVolumeSetOwnerResponse(omResponse.build());
    }
```
- Replay
```java
      // Check if this transaction is a replay of ratis logs.
      // If this is a replay, then the response has already been returned to
      // the client. So take no further action and return a dummy
      // OMClientResponse.
      if (isReplay(ozoneManager, omVolumeArgs, transactionLogIndex)) {
        LOG.debug("Replayed Transaction {} ignored. Request: {}",
            transactionLogIndex, setVolumePropertyRequest);
        return new OMVolumeSetOwnerResponse(createReplayOMResponse(omResponse));
      }
```
- newOwner is the same as oldOwner (added in this PR)
```java
      // Return OK immediately if newOwner is the same as oldOwner.
      if (oldOwner.equals(newOwner)) {
        LOG.warn("Volume '{}' owner is already user '{}'.", volume, oldOwner);
        omResponse.setStatus(OzoneManagerProtocolProtos.Status.OK)
          .setMessage(
            "Volume '" + volume + "' owner is already '" + newOwner + "'.")
          .setSuccess(false);
        return new OMVolumeSetOwnerResponse(omResponse.build());
      }
```
- IOException
```java
    } catch (IOException ex) {
      exception = ex;
      omClientResponse = new OMVolumeSetOwnerResponse(
          createErrorOMResponse(omResponse, exception));
    } finally {
```

Reply:
Though in all four cases the success flag is set to false, rendering the condition around `checkStatusNotOK()` pointless.

Reply:
Just BTW in a discussion with @bharatviswa504 we concluded that the reason of setting success to `false` for "newOwner equals to oldOwner" case is that in [`addAcl`](https://github.com/apache/hadoop-ozone/blob/ebc616061da6f337626ee85e4282230aedfd748a/hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/shell/acl/AclOption.java#L58-L62) we have some logic to print different results on the client depending on the success flag. Though the success flag of `setOwner` isn't utilized at the moment, we might be able to use it later.

Reply:
> Though in all four cases the success flag is set to false, rendering the condition around `checkStatusNotOK()` pointless.

Maybe we should just remove the call to `checkStatusNotOK()` in this constructor now that we established this new use case. What do you think?

Reply:
Just to add more info, we set the success flag and also there is a response which we return that is used by the end client which has a response flag which is set based on the result of the operation. 

```
  OMClientResponse onSuccess(OMResponse.Builder omResponse,
      OmBucketInfo omBucketInfo, boolean operationResult) {
    omResponse.setSuccess(operationResult);
    omResponse.setAddAclResponse(AddAclResponse.newBuilder()
         .setResponse(operationResult));
```

But in OmVolumeSetOwner case, there is no such flag. If we can introduce a flag in SetVolumePropertyResponse, we can use the success flag.

      omResponse.setSetVolumePropertyResponse(
          SetVolumePropertyResponse.newBuilder().build());

Reply:
> Just to add more info, we set the success flag and also there is a response which we return that is used by the end client which has a response flag which is set based on the result of the operation.
> 
> ```
>   OMClientResponse onSuccess(OMResponse.Builder omResponse,
>       OmBucketInfo omBucketInfo, boolean operationResult) {
>     omResponse.setSuccess(operationResult);
>     omResponse.setAddAclResponse(AddAclResponse.newBuilder()
>          .setResponse(operationResult));
> ```
> 
> But in OmVolumeSetOwner case, there is no such flag. If we can introduce a flag in SetVolumePropertyResponse, we can use the success flag.
> 
> ```
>   omResponse.setSetVolumePropertyResponse(
>       SetVolumePropertyResponse.newBuilder().build());
> ```

Good idea we can add `optional bool response = 1;` in [here](https://github.com/apache/hadoop-ozone/blob/15db251f16236228c6596253dab6946494fc8f5b/hadoop-ozone/common/src/main/proto/OzoneManagerProtocol.proto#L425-L427) in another jira.

Reply:
Discussed with @xiaoyuyao and we agree the comment don't need to be changed at the moment.

Resolving this conversation. Will commit shortly.